docs/tutorials/on-device-training/ios-app.md

```diff
@@ -0,0 +1,987 @@
+    - [ContentView](#contentview)
+- [Running the iOS application](#running-the-ios-application)
+- [Conclusion](#conclusion)
+
+
+## Prerequisites
+
+To follow along with this tutorial, you should have a basic understanding of machine learning and iOS development. You should also have the following installed on your machine:
+
+- Python development environment to prepare the training artifacts
+- [Xcode](https://developer.apple.com/xcode/)
+- [CocoaPods](https://cocoapods.org/)
+
+> **Note:**  The entire iOS application is also made available on the [`onnxruntime-training-examples`](https://github.com/microsoft/onnxruntime-training-examples/tree/master/on_device_training/mobile/ios) GitHub repository. You can clone the repository and follow along with the tutorial.
+
+## Generating the training artifacts
+
+1. ### Export the model to ONNX.
+
+    We will start with a pre-trained model from *HuggingFace* and export it to ONNX. The `wav2vec` model has been pretrained on `VoxCeleb1`, which has more than 1000 categories. For our task, we only need to classify audio in 2 classes. So, we change the last layer of the model to output 2 classes. We will use the `transformers` library to load the model and export it to ONNX.
+
+    ```python
+    from transformers import Wav2Vec2ForSequenceClassification, AutoConfig
+    import torch
+
+    # load config from the pretrained model
+    config = AutoConfig.from_pretrained("superb/wav2vec2-base-superb-sid")
+    model = Wav2Vec2ForSequenceClassification.from_pretrained("superb/wav2vec2-base-superb-sid")
+
+    # modify last layer to output 2 classes
+    model.classifier = torch.nn.Linear(256, 2)
+
+    #export model to ONNX
+    dummy_input = torch.randn(1, 160000, requires_grad=True)
+    torch.onnx.export(model, dummy_input, "wav2vec.onnx",input_names=["input"], output_names=["output"],
+                    dynamic_axes={"input": {0: "batch"}, "output": {0: "batch"}})
+    ```
+2. ### Define the trainable and non trainable parameters
+
+    ```python
+    import onnx
+
+    # load the onnx model
+    onnx_model = onnx.load("wav2vec.onnx")
+
+    # Define the parameters that require gradients to be computed (trainable parameters) and
+    # those that don't (frozen/non-trainable parameters)
+    requires_grad = ["classifier.weight", "classifier.bias"]
+    frozen_params = [
+        param.name
+        for param in onnx_model.graph.initializer
+        if param.name not in requires_grad
+    ]
+    ```
+3. ### Generate the training artifacts. 
+    We will use the `CrossEntropyLoss` loss and the `AdamW` optimizer for this tutorial. More details around artifact generation can be found [here](../../../docs/api/python/on_device_training/training_artifacts.html).
+
+    Since, the model also outputs logits and hidden states, we will use `onnxblock` to define a custom loss function that extracts logits from the model output and passes it to the `CrossEntropyLoss` function.
```

Comment:
```suggestion
    Since the model also outputs logits and hidden states, we will use `onnxblock` to define a custom loss function that extracts logits from the model output and passes it to the `CrossEntropyLoss` function.
```